frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -226,8 +243,27 @@ const edgeDragSourceSpec = (
 
 const createConnectorCallback = (serviceBinding: boolean) => (
   source: Node,
-  target: Node,
+  target: Node | Graph,
 ): any[] | null => {
+  const onKebabOptionClick = (option: KebabOption) => {
+    if (option.callback) {
+      option.callback();
+    }
+    if (option.href) {
+      const sourceObj = _.get(source.getData(), ['resources', 'obj'], null);
```

Comment:
You can use optional chaining here.

Reply:
should we allow creation of resources through this flow if sourceObj is null or undefined. since it will reduce the flow to a simple add-resource flow?

Reply:
In which case the sourceObj will be undefined though? Dragging sourceObj connector and on releasing we will see the menu to create the service.

Reply:
the sourceObj 

> @nemesis09 will there be a + icon for a connector within a group, currently it's arrow
> 
> <img alt="Screenshot 2020-01-24 at 5 35 41 PM" width="554" src="https://user-images.githubusercontent.com/5129024/73068034-36b25d00-3ed0-11ea-8747-7cc9162e7727.png">

Done

Reply:
The only time i saw an error was when component did not yet mount. other than that, I think the source will always be there.

Reply:
updated the code with suggested change

Reply:
If you have seen the error probably do not allow redirect to create flow if sourceObj is undefined

Reply:
the new changes takes care of it